components/inputs/input-time.js

```diff
@@ -270,19 +264,28 @@ class InputTime extends FocusMixin(LabelledMixin(SkeletonMixin(FormElementMixin(
 			this._formattedValue = formatTime(time);
 		}
 
-		const hiddenContent = this.shadowRoot.querySelector('.d2l-input-time-hidden-content');
+		const updateHiddenContentWidth = () => {
+			const hiddenContent = this.shadowRoot.querySelector('.d2l-input-time-hidden-content');
+			const width = Math.ceil(parseFloat(getComputedStyle(hiddenContent).getPropertyValue('width')));
+			this._hiddenContentWidth = `${width}px`;
+			/** @ignore */
+			this.dispatchEvent(new CustomEvent(
+				'd2l-input-time-hidden-content-width-change',
+				{ bubbles: true, composed: false }
+			));
+		};
+
 		this.addEventListener('d2l-localize-resources-change', async() => {
 			await this.updateComplete;
 			this._formattedValue = formatTime(getDateFromISOTime(this.value));
 			INTERVALS.clear();
 			this.requestUpdate();
-			this.updateComplete.then(() => this._onResize(hiddenContent));
+			this.updateComplete.then(() => updateHiddenContentWidth());
 		});
 
 		await (document.fonts ? document.fonts.ready : Promise.resolve());
 
-		this._hiddenContentResizeObserver = new ResizeObserver(() => this._onResize(hiddenContent));
-		this._hiddenContentResizeObserver.observe(hiddenContent);
+		updateHiddenContentWidth();
```

Comment:
I'm wondering if this is calculating too early?  And the ResizeObserver was added so there'd be a delay, and it would run _next_ time we resize?

Reply:
Thanks, the history on why it was introduced was something I wanted to investigate and test. Turns out that's exactly why -- when the components are `display: none` initially, the width returns as "auto" which can't be parsed. The observer fixes that.

I've updated it slightly so that if it can't get a width right away, THEN it adds the observer. This still fixes the tests in Insights but if things were initially hidden there could still be issues so I'm going to separately look into the other observers and see if they can be done a different way.